boot/bootutil/include/bootutil/image.h

```diff
@@ -101,6 +112,7 @@ struct flash_area;
 #define IMAGE_TLV_DEPENDENCY        0x40   /* Image depends on other image */
 #define IMAGE_TLV_SEC_CNT           0x50   /* security counter */
 #define IMAGE_TLV_BOOT_RECORD       0x60   /* measured boot record */
+#define IMAGE_TLV_COMP_SIZE         0x70   /* Compressed image size */
```

Comment:
Additionally, it is a little unclear what the sizes mean. Does this mean the has the uncompressed size? How would we find the TLV if the actual data stored is different than the size in the header?

Reply:
For some reason I thought it went through looking for the magic bytes, apparently not... Will have a rethink

Reply:
Have updated the PR to instead have 3 TLVs, the hash and signature would be equivalent to the compressed image fields but on the decompressed image data instead (this is so that the data can be validated before decompressing it)